simulariumio/cellpack/cellpack_converter.py

```diff
@@ -0,0 +1,347 @@
+        z_size = bb[1][2] - bb[0][2]
+        return [
+            x_size,
+            y_size,
+            z_size,
+        ]
+
+    @staticmethod
+    def _get_euler_from_matrix(data_in, handedness):
+        rotation_matrix = [data_in[0][0:3], data_in[1][0:3], data_in[2][0:3]]
+        if handedness == HAND_TYPE.LEFT:
+            euler = R.from_matrix(rotation_matrix).as_euler("ZYX", degrees=False)
+            return [euler[0], euler[1], -euler[2]]
+        else:
+            return R.from_matrix(rotation_matrix).as_euler("XYZ", degrees=False)
+
+    @staticmethod
+    def _get_euler_from_quat(data_in, handedness):
+        if handedness == HAND_TYPE.LEFT:
+            euler = R.from_quat(data_in).as_euler("ZYX", degrees=False)
+            return [euler[0], euler[1], -euler[2]]
+        else:
+            return R.from_quat(data_in).as_euler("XYZ", degrees=False)
+
+    @staticmethod
+    def _is_matrix(data_in):
+        if isinstance(data_in[0], list):
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def _get_euler(data_in, handedness) -> np.array:
+        if CellpackConverter._is_matrix(data_in):
+            return CellpackConverter._get_euler_from_matrix(data_in, handedness)
+        else:
+            return CellpackConverter._get_euler_from_quat(data_in, handedness)
+
+    @staticmethod
+    def _unpack_curve(
+        data,
+        time_step_index: int,
+        ingredient_name: str,
+        index: int,
+        agent_id: int,
+        result: AgentData,
+        scale_factor: float,
+        box_center: np.array,
+    ):
+        curve = "curve" + str(index)
+        result.positions[time_step_index][agent_id] = [0, 0, 0]
+        result.rotations[time_step_index][agent_id] = [0, 0, 0]
+        result.viz_types[time_step_index][agent_id] = 1001
+        result.n_agents[time_step_index] += 1
+        result.types[time_step_index].append(ingredient_name)
+        result.unique_ids[time_step_index][agent_id] = agent_id
+        r = (
+            data["encapsulatingRadius"] * scale_factor
+            if ("encapsulatingRadius" in data)
+            else 1
+        )
+        result.radii[time_step_index][agent_id] = r
+        result.n_subpoints[time_step_index][agent_id] = len(data[curve])
+        scaled_control_points = (
+            np.array(data[curve]) - np.array(box_center)
+        ) * scale_factor
+        for i in range(len(scaled_control_points)):
+            result.subpoints[time_step_index][agent_id][i] = scaled_control_points[i]
+
+    @staticmethod
+    def _unpack_positions(
+        data,
+        time_step_index: int,
+        ingredient_name: str,
+        index: int,
+        agent_id,
+        result: AgentData,
+        scale_factor: float,
+        box_center: np.array,
+        handedness: HAND_TYPE,
+        comp_id=0,
+    ):
+        position = data["results"][index][0]
+        offset = np.array([0, 0, 0])
+        # TODO: use offset from data
```

Comment:
should this be done now?

Reply:
I'm going to write a better comment, but basically this offset and comp_id both have to do with PDBs on the surface of membranes and I would prefer to get this v1.0 into simulariumio, so I can then start using it in cellpack instead of having two different conversion scripts, and then do an update once I have tested more complex recipes. I want to use this to be able to write directly to simularium files for faster viewing of results